### DISCUSSION TITLE
Slow page navigation?

### DISCUSSION THREAD

**Discussion by albingroen** [remix-run/remix#3049] [Q&A] [answered]
I've now built my first Remix app, and in general I'm really impressed with the framework.

The app is now hosted on Vercel, and the Postgres database on Heroku.

The problem I'm having is that page navigation is quite sluggish. Is this expected in production or am I doing something wrong?

**Comment by albingroen** [comment]
Ok it's defintely faster when deploying to fly.io at least

**Comment by sergiodxa** [comment, marked as answer]
The reason for the slow navigation is most likely because your app (Vercel) and your DB (Heroku) are not in the same server, so there’s a latency, if they are also in different regions the latency will increase making it even slower.

The reason why Fly.io feels faster is because both app and DB are deployed together so the latency is minimal and almost imperceptible.

**Reply by albingroen** [reply]
In my case, I wouldn't say it's imperceptible, but it's definitely faster

**Reply by sergiodxa** [reply]
I was talking on the latency between your server and your DB, you will still have latency between your computer and your server
